clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy

```diff
@@ -265,11 +266,29 @@ class BasicGoogleDeployHandler implements DeployHandler<BasicGoogleDeployDescrip
         Backend sourceBackend = backendService.backends.find { Backend backend ->
           Utils.getLocalName(backend.group) == description.source.serverGroupName
         }
-        if (!sourceBackend) {
+        if (description.source.serverGroupName && !sourceBackend) {
```

Comment:
Even though `source` is assigned a value at initialization of the containing `description`, someone could still make it explicitly null in the request. So let's use `?.` between `description` and `source`. Here, and the couple of other spots where `source` is dereferenced.